opensrp-reveal/src/main/java/org/smartregister/reveal/view/ListTasksActivity.java

```diff
@@ -293,9 +332,28 @@ public void displayNotification(String message) {
     }
 
     @Override
-    public void openCardView(String structureId, String taskIdentifier, String businessStatus) {
-        Toast.makeText(this, String.format("Opening Card View for Structure %s and task %s",
-                structureId, taskIdentifier), Toast.LENGTH_SHORT).show();
+    public void openCardView(CardDetails cardDetails) {
+        TextView tvSprayStatus = findViewById(R.id.spray_status);
```

Comment:
avoid calling findViewById every time a structure is clicked, Call findViewById once and store the references to the textviews